lib/Server/Starter.pm

```diff
@@ -389,7 +389,9 @@ sub start_server {
             my $kill_old_delay = defined $ENV{KILL_OLD_DELAY} ? $ENV{KILL_OLD_DELAY} : $ENV{ENABLE_AUTO_RESTART} ? 5 : 0;
             if ($kill_old_delay != 0) {
                 print STDERR "sleeping $kill_old_delay secs before killing old workers\n";
-                sleep $kill_old_delay;
+                while ($kill_old_delay > 0) {
+                    $kill_old_delay -= sleep $kill_old_delay;
```

Comment:
Could you adjust the code so that `$kill_old_delay` will be decremented by one in case `sleep` returns zero?

My understanding is that the seconds returned by `sleep` is rounded down. And my fear that the loop might run forever if signals get delivered more than once a second.